### PR TITLE
CatsAndMods bugs

### PR DESCRIPTION
**Pull request by markd315**
This fixed the backend deletion and menus for catsandmods deletion.

As of 3:32PM 11/1/2017 these are the tickets we might need tackled for the next PR:
4) Fix the list items view so that the cats and mods display correctly, ideally with the colors (lower priority)
6) Users should be created with the role they signed up for.

### CODE REVIEW

File: modules/items/client/controllers/admin/list-categories.client.controller.js [hunk 1]
```diff
@@ -5,21 +5,18 @@
...
+  function ItemsAdminCategoriesController(CategoriesService, $window, $state, $scope, Notification) {
     var vm = this;
 
     vm.categories= CategoriesService.query();
-    vm.remove= remove;
-    function remove(cat) {
-      if ($window.confirm('Are you sure you want to create?')) {
-        cat.$rem(function () {
-          $state.go('items.list');
-          Notification.success({ message: '<i class="glyphicon glyphicon-ok"></i> Item created successfully!' });
```
Comment (RyanClementsHax) [inline modules/items/client/controllers/admin/list-categories.client.controller.js:19]: we should keep this notification

File: modules/items/client/controllers/admin/cats.client.controller.js [hunk 2]
```diff
@@ -13,20 +13,10 @@
     vm.cat = cat;
     vm.authentication = Authentication;
     vm.form = {};
-    vm.remove = remove;
     vm.create = create;
     vm.list = list;
 
     // Remove existing Item
```
Comment (RyanClementsHax) [inline modules/items/client/controllers/admin/cats.client.controller.js]: if we are deleting the function, we also need to delete the comment

File: modules/items/client/config/items.client.routes.js [hunk 3]
```diff
@@ -20,25 +20,6 @@
         controller: 'ItemsListController',
         controllerAs: 'vm'
       })
-      .state('items.addCats', {
```
Comment (RyanClementsHax) [inline modules/items/client/config/items.client.routes.js:23]: these seem like important states

File: modules/items/client/config/items.client.menus.js [hunk 4]
```diff
@@ -21,15 +21,5 @@
       state: 'items.list',
       roles: ['admin', 'superta', 'ta', 'technician']
     });
-    menuService.addSubMenuItem('topbar', 'items', {
```
Comment (RyanClementsHax) [inline modules/items/client/config/items.client.menus.js:24]: how are we going to access these pages?

File: modules/items/client/controllers/admin/list-modules.client.controller.js [hunk 5]
```diff
@@ -5,16 +5,18 @@
...
 
-  function ItemsAdminModulesController(ItemsService, ModulesService) {
+  function ItemsAdminModulesController(ModulesService, $window, $state, $scope, Notification) {
     var vm = this;
 
     vm.modules = ModulesService.query();
 
-    vm.deleteModule = function(module){
-    	vm.modules.splice(vm.modules.indexOf(module), 1);
+    vm.remove = function(module){
```
Comment (RyanClementsHax) [inline modules/items/client/controllers/admin/list-modules.client.controller.js:15]: we want to do a notification here too

File: modules/items/server/controllers/items.server.controller.js [hunk 6]
```diff
@@ -38,7 +38,9 @@ exports.createCategory = function (req, res) {
 };
 
 exports.deleteCategory = function(req, res) {
-  var toDelete = req.body;
+  console.log(req);
```
Comment (RyanClementsHax) [inline modules/items/server/controllers/items.server.controller.js]: morgan will auto log stuff for us, we dont need these console statements

**Review by RyanClementsHax — DISMISSED** [review]
I'm a little confused. It seems like we are removing important code.

File: modules/items/client/controllers/admin/list-modules.client.controller.js (hunk 5 above)
Comment (markd315) [reply modules/items/client/controllers/admin/list-modules.client.controller.js:15]: There is a notification already.

File: modules/items/client/controllers/admin/cats.client.controller.js (hunk 2 above)
Comment (markd315) [reply modules/items/client/controllers/admin/cats.client.controller.js]: Yep

File: modules/items/client/controllers/admin/list-categories.client.controller.js (hunk 1 above)
Comment (markd315) [reply modules/items/client/controllers/admin/list-categories.client.controller.js:19]: There's still a notification.

File: modules/items/client/config/items.client.routes.js (hunk 3 above)
Comment (markd315) [reply modules/items/client/config/items.client.routes.js:23]: They're not.

File: modules/items/client/config/items.client.menus.js (hunk 4 above)
Comment (markd315) [reply modules/items/client/config/items.client.menus.js:24]: We totally are, but not from the TA perspective, only the admin is entitled access per the answer to our questioning.

File: modules/items/server/controllers/items.server.controller.js (hunk 6 above)
Comment (markd315) [reply modules/items/server/controllers/items.server.controller.js]: Good catch.

**Review by kevdaci — APPROVED** [review]
Everything looks go to me.

**Pull request merged**